src/kv/lazy-store.ts

```diff
@@ -0,0 +1,202 @@
+import {RWLock} from '../rw-lock';
+import type {Read, Store, Value, Write} from './store';
+import {deleteSentinel, WriteImplBase} from './write-impl-base';
+import {getSizeOfValue} from '../get-size-of-value';
+
+export class LazyStore implements Store {
```

Comment:
Would `MemStore` go away? It seems like `LazyStore` is the same as `MemStore` if `shouldBePinned` always returns true.

Reply:
(if so maybe the right name for this thing is `MemStore` :-)).

Reply:
Yes MemStore goes away.  I imagined it went away in a future change, but you are right, I could replace it with a LazyStore with shouldBePinned => true.  I like that.  I'll make it part of this change.

Reply:
Actually I am going to wait till a later integration change to get rid of MemStore.  LazyStore needs a base Store, and I don't want to make the change to always create an IDBStore as part of this change.  At that time, we can also rename LazyStore to MemStore.

Reply:
It seems like MemStore can be removed and we can use kv.TestStore in all the tests